src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Workable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        switch (type.getSimpleName()) {
+            case "Truck" :
+                return new TruckProducer().get();
+            case "Bulldozer" :
+                return new BulldozerProducer().get();
+            case "Excavator" :
+                return new ExcavatorProducer().get();
```

Comment:
https://mate-academy.github.io/jv-program-common-mistakes/java-core/pecs/pecs.html#consider-creating-a-local-variable-in-your-implementation-of-getallclass-type-method